vsphere/internal/helper/hostsystem/host_system_helper.go

```diff
@@ -98,6 +107,19 @@ func NameOrID(client *govmomi.Client, id string) string {
 	return name
 }
 
+// HostInMaintenance checks a HostSystem's maintenance mode and returns true if the
+// the host is in maintenance mode.
+func HostInMaintenance(host *object.HostSystem) (bool, error) {
+	var hostObject mo.HostSystem
+	err := host.Properties(context.Background(), host.Reference(), nil, &hostObject)
```

Comment:
With [this](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/internal/helper/hostsystem/host_system_helper.go#L62), you don't need to worry about supplying the context and the provider default timeout is used automatically.

Reply:
👍